### PR TITLE
Fixed #34111 -- Made test runner with --debug-sql format SQL queries.

### PR DESCRIPTION
**Pull request by Giebisch**
Solves Ticket [#34111](https://code.djangoproject.com/ticket/34111)

SQL Debug text before: 

```
----------------------------------------------------------------------
(0.000) SELECT "aggregation_publisher"."id", "aggregation_publisher"."name", "aggregation_publisher"."num_awards", "aggregation_publisher"."duration", EXISTS(SELECT 1 AS "a" FROM "aggregation_book" U0 WHERE U0."publisher_id" = ("aggregation_book"."publisher_id") LIMIT 1) AS "books_exists", COUNT("aggregation_book"."id") AS "books_count" FROM "aggregation_publisher" LEFT OUTER JOIN "aggregation_book" ON ("aggregation_publisher"."id" = "aggregation_book"."publisher_id") GROUP BY "aggregation_publisher"."id", EXISTS(SELECT 1 AS "a" FROM "aggregation_book" U0 WHERE U0."publisher_id" = ("aggregation_book"."publisher_id") LIMIT 1); args=(1, 1); alias=default

----------------------------------------------------------------------
```

and now looks like this:

```
----------------------------------------------------------------------
(0.000)
SELECT "aggregation_publisher"."id",
       "aggregation_publisher"."name",
       "aggregation_publisher"."num_awards",
       "aggregation_publisher"."duration",
       EXISTS
  (SELECT 1 AS "a"
   FROM "aggregation_book" U0
   WHERE U0."publisher_id" = ("aggregation_book"."id")
   LIMIT 1) AS "books_exists",
       COUNT("aggregation_book"."id") AS "books_count"
FROM "aggregation_publisher"
LEFT OUTER JOIN "aggregation_book" ON ("aggregation_publisher"."id" = "aggregation_book"."publisher_id")
GROUP BY "aggregation_publisher"."id",
         "aggregation_publisher"."name",
         "aggregation_publisher"."num_awards",
         "aggregation_publisher"."duration",
         "aggregation_book"."id";

args=(1,);

ALIAS=DEFAULT (0.000)
SELECT "aggregation_publisher"."id",
       "aggregation_publisher"."name",
       "aggregation_publisher"."num_awards",
       "aggregation_publisher"."duration"
FROM "aggregation_publisher";

args=();

ALIAS=DEFAULT
----------------------------------------------------------------------
Ran 1 test in 0.020s

FAILED (failures=1)
Destroying test database for alias 'default'...
Total database setup took 0.093s
  Creating 'default' took 0.093s
Total database teardown took 0.000s
Total run took 0.272s
```

### CODE REVIEW

File: tests/test_runner/test_debug_sql.py [hunk 1]
```diff
@@ -74,14 +74,14 @@ def _test_output(self, verbosity):
...
-        full_output = self._test_output(1)
+        full_output = self._test_output(1).replace("\n", " ")
         for output in self.expected_outputs:
             self.assertIn(output, full_output)
         for output in self.verbose_expected_outputs:
             self.assertNotIn(output, full_output)
 
     def test_output_verbose(self):
-        full_output = self._test_output(2)
+        full_output = self._test_output(2).replace("\n", " ")
```
Comment (charettes) [inline tests/test_runner/test_debug_sql.py:84]: Thanks for the patch @Giebisch, I think we should adjust the `expected_outputs` instead to be properly formatted in order to ensure formatting is actually used.

https://github.com/django/django/blob/78657a201308a0b53661e2b267914407a9fd7fa1/tests/test_runner/test_debug_sql.py#L90-L111

https://github.com/django/django/blob/78657a201308a0b53661e2b267914407a9fd7fa1/tests/test_runner/test_debug_sql.py#L124-L133

The adjustments made make the tests pass with and without the proposed changes.

**Comment by shangxiao** [comment]
@Giebisch ♥️ Thanks for adding this – I would love to see this.

@charettes @felixxm Crazy idea here, but is this something we can have for logging SQL as well - I don't want sqlparse to be a dependency but we could optionally parse the logged sql only if sqlparse is available to import.  This would *greatly* help with my development process.

**Comment by charettes** [comment]
> I don't want sqlparse to be a dependency

Well it already is :)

https://github.com/django/django/blob/5ec64fa481892747ee5ce7ec13584cc4fe53b857/setup.cfg#L43

> Crazy idea here, but is this something we can have for logging SQL as well

I'm not sure if this needs to live in core given the performance impact it would have but it feels like the best way to achieve it would be to have a formatter for the `django.db.backends` logger so it can be removed if it has a noticeable impact on performance.

This is something you can do even today without altering core by pointing `settings.LOGGING['formatters']['django.db.backends'] = {'()': 'your.SQLParseFormatter'}` that retrieves `record.extra['sql']`.

**Review by pauloxnet — APPROVED** [review]

**Comment by felixxm** [comment]
@Giebisch Thanks for this patch :+1: Welcome aboard :boat: 

I noticed that it crashes with `sqlparse==0.2.2`:
```
./runtests.py test_runner.test_debug_sql
Testing against Django installed in '/home/felixx/repo/django/django' with up to 8 processes
Found 3 test(s).
System check identified no issues (0 silenced).
EE.
======================================================================
ERROR: test_output_normal (test_runner.test_debug_sql.TestDebugSQL)
----------------------------------------------------------------------
...
RuntimeError: generator raised StopIteration

======================================================================
ERROR: test_output_verbose (test_runner.test_debug_sql.TestDebugSQL)
----------------------------------------------------------------------
Traceback (most recent call last):
...
  File "/lib/python3.8/site-packages/sqlparse/filters/reindent.py", line 116, in _process_parenthesis
    with offset(self, self._get_offset(first) + 1):
  File "/lib/python3.8/site-packages/sqlparse/filters/reindent.py", line 41, in _get_offset
    raw = u''.join(map(text_type, self._flatten_up_to_token(token)))
RuntimeError: generator raised StopIteration

----------------------------------------------------------------------
Ran 3 tests in 0.276s

FAILED (errors=2)
```
so I bumped the minimum supported version of ``sqlparse`` to 0.2.3.

**Pull request merged**